cmd/gnodev/test.go

```diff
@@ -54,6 +54,18 @@ func testApp(cmd *command.Command, args []string, iopts interface{}) error {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	// go.mod
+	modPath := filepath.Join(tempdirRoot, "go.mod")
+	out, err := makeTestGoMod()
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(modPath, []byte(out), 0o644)
+	if err != nil {
+		return fmt.Errorf("write .mod file: %w", err)
+	}
+
```

Comment:
What do you think about extracting this into a separate method?

You can easily move this write logic to `makeTestGoMod`, as `makeTestGoMod` currently only generates a string, not the actual `go.mod` file on the disk.

Reply:
Resolving this convo as you've rewritten the implementation